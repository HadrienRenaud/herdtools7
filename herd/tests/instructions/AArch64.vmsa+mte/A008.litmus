AArch64 A008
{
 [x]=1;
 [PTE(x)]=(valid:0);
 0:X0=x; 0:X1=x:red;
 1:X1=(oa:PA(x),valid:1);
 1:X0=PTE(x);
}
 P0           | P1          ;
L0:           | STR X1,[X0] ;
 STZG X1,[X0] |             ;
exists(fault(P0:L0,x,MMU:Translation) /\ [x]=0)
