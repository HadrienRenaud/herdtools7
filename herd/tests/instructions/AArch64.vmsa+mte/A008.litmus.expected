Test A008 Allowed
States 3
[x]=0;  ~Fault(P0:L0,x,MMU:Translation);
[x]=0; Fault(P0:L0,x,MMU:Translation);
[x]=1; Fault(P0:L0,x,MMU:Translation);
Ok
Witnesses
Positive: 1 Negative: 4
Flag combining-vmsa-and-memtag-is-not-supported
Condition exists (fault(P0:L0,x,MMU:Translation) /\ [x]=0)
Observation A008 Sometimes 1 4
Hash=476835ada4ec0e0c73096d09264acc89

